### ISSUE TITLE
Opacity for drawn primitives

### ISSUE THREAD

**Issue by tgrosh** [bjorn/tiled#855]
When drawing primitives on the Object Layers, they are drawn with only partial opacity in the Tiled window. I know there is an opacity property, but I would like to be able to see the primitives drawn with full opacity, or at least match the opacity I have set.

For example, if I have 2 layers, and I draw a bunch of primitives on each of them. I can see all of the primitives on the screen through one another because they are drawn in the editor window with only partial opacity. This is useful sometimes, but there are times that I want to see them as they would be rendered downstream by my game (which would use the opacity property from the property pane, which is defaulted to 1).

It would be great if there were a keybind to switch back and forth between the current behavior (arbitrary partial opacity) and the opacity set at the layer level.

**Comment by bjorn** [comment]
Hmm, the primitives you're talking about where never meant to be displayed in a game as-is. They are there for annotating maps with meta-information. As such, no real options exist for changing how they are rendered except for the color thing which is only meant to make it easier to distinguish different types of objects.

It would only be a small code change to make them fully opaque, but this isn't nice for many other users who rely on being able to see things through these annotation objects. However, it is something that could be added as a toggle as you suggested.

**Comment by wayfu** [comment]
I too am using polygons and circles as my primary mapping tools ;) I didn't even know they were meant for annotations. Never written in them even once! :)

**Comment by tgrosh** [comment]
I commited a change on https://github.com/tgrosh/tiled that makes the objects in the object layers use the layer opacity. Works well, but I could not follow how to tie this into preferences. Ideally, would like to see a "Objects use Layer Opacity" option on the View menu that turns this on or off.

**Comment by bjorn** [comment]
Note that objects already use the layer opacity without that part of your change. This is because the opacity of the `ObjectGroupItem` is already synchronized with the opacity of the layer, which affects all child `MapObjectItem` instances. See https://github.com/tgrosh/tiled/blob/master/src/tiled/mapscene.cpp#L402

So the option in the preferences should just be called something like "Draw opaque object shapes".

**Comment by tgrosh** [comment]
I certainly defer to your knowledge of whats best for the application. I know without my change, the opacity of the shapes never gets better than .4 (the current opacityFactor). I want the opacity of the layer to affect it, but at the moment its not allowed to become completely opaque.

**Comment by bjorn** [comment]
Yes, I was just saying that the only change you would have needed to make was the one you did to the line `brushColor.setAlpha(255);`. The `mapobjectitem.cpp` part of your change was unnecessary, because the objects are already using the alpha value of the object group when they are being drawn.

**Comment by tgrosh** [comment]
Absolutely correct. I removed all changes from mapobjectitem.cpp and basically just removed the line that sets alpha in the orthogonal renderer.
